usaspending_api/search/v2/views/spending_over_time.py

```diff
@@ -142,19 +150,67 @@ def parse_elasticsearch_bucket(self, bucket: dict) -> dict:
+
+        # extract the categorical breakdown
+        categories_breakdown = bucket["group_by_category"]["buckets"]
+
+        # Initialize a dictionary to hold the query results for each obligation type
+        category_dictionary = {
+            "Contract_Obligations": 0,
+            "Direct_Obligations": 0,
+            "Grant_Obligations": 0,
+            "Idv_Obligations": 0,
+            "Loan_Obligations": 0,
+            "Other_Obligations": 0,
+        }
+
+        for category in categories_breakdown:
+            if category["key"] == "contract":
+                category_dictionary["Contract_Obligations"] = category.get("sum_as_dollars", {"value": 0})["value"]
+            elif category["key"] == "direct payment":
+                category_dictionary["Direct_Obligations"] = category.get("sum_as_dollars", {"value": 0})["value"]
+            elif category["key"] == "grant":
+                category_dictionary["Grant_Obligations"] = category.get("sum_as_dollars", {"value": 0})["value"]
+            elif category["key"] == "idv":
+                category_dictionary["Idv_Obligations"] = category.get("sum_as_dollars", {"value": 0})["value"]
+            elif category["key"] == "loans":
+                category_dictionary["Loan_Obligations"] = category.get("sum_as_dollars", {"value": 0})["value"]
+            elif category["key"] == "other":
+                category_dictionary["Other_Obligations"] += category.get("sum_as_dollars", {"value": 0})["value"]
+            elif category["key"] == "insurance":
+                category_dictionary["Other_Obligations"] += category.get("sum_as_dollars", {"value": 0})["value"]
 
         if self.group == "quarter":
             quarter = (key_as_date.month - 1) // 3 + 1
             time_period["quarter"] = str(quarter)
         elif self.group == "month":
             time_period["month"] = str(key_as_date.month)
 
-        aggregated_amount = bucket.get("sum_as_dollars", {"value": 0})["value"]
-        return {"aggregated_amount": aggregated_amount, "time_period": time_period}
+        aggregated_amount = sum(category_dictionary[item] for item in category_dictionary)
+
+        response_object = {
+            "aggregated_amount": aggregated_amount,
+            "time_period": time_period,
+        }
+
+        # Update the response object with the contents of category_dictionary
+        response_object.update(category_dictionary)
+
+        return response_object
 
     def build_elasticsearch_result(self, agg_response: AggResponse, time_periods: list) -> list:
+        """
+        In this function we are justing taking the elasticsearch aggregate response and looping through the
+        buckets to create a results object for each time interval
+        """
         results = []
         min_date, max_date = min_and_max_from_date_ranges(time_periods)
         fiscal_date_range = generate_fiscal_date_range(min_date, max_date, self.group)
```

Comment:
I think it's a little tricky that this function and returned variable are referring to fiscal dates even if the calendar year group are used. 

Think about line 225:
```
time_period = {"calendar_year": str(fiscal_date["calendar_year"])}
```

I was pretty confused to see the `calendar_year` coming from a `fiscal_date` dictionary until I remembered that the `generate_fiscal_date_range()` can also generate non fiscal date range (calendar ranges) as well.

Reply:
I have left a detailed comment regarding this.... I'm not sure if we really want to create an entirely separate function for just creating a non-fiscal date range... I think comment works just fine.

Reply:
The information in this comment is great! I think it would be better suited in the pydoc comment for the function itself rather than being here before the usage of the function. 

Also, I understand not wanting to create a second function to handle non fiscal years, but I think renaming the function and output variable (`fiscal_date_range`) might make sense. Maybe just `generate_date_range()` and `date_range` instead?